state/execution.go

```diff
@@ -115,6 +115,25 @@ func (blockExec *BlockExecutor) CreateProposalBlock(
 	if err != nil {
 		panic(err)
 	}
+
+	// remove invalid txs from mempool
+	var invalidTxs = make([]types.Tx, len(resp.InvalidTxs))
+	var deliverTxResponses = make([]*abcix.ResponseDeliverTx, len(resp.InvalidTxs))
+	for i, tx := range resp.InvalidTxs {
+		copy(invalidTxs[i], tx)
+		deliverTxResponses[i].Code = 1
+	}
+	err = blockExec.mempool.Update(
```

Comment:
1. the first comment of that function says lock has to be acquired.
1. seems we don't need precheck and postcheck?

Reply:
also this seems will update mempool's internal state, which may cause unexpected situations. can we only do one thing: remove those invalid txs?

Reply:
ok I will fix it~